src/main/java/com/lob/client/AsyncLobClient.java

```diff
@@ -495,10 +498,22 @@ private BoundRequestBuilder post(final String resourceUrl, final HasLobParams ha
         final SettableFuture<T> guavaFut = SettableFuture.create();
         try {
             final Optional<String> apiVersionOpt = Lob.getApiVersion();
+            String wrapperVersion;
+
             if (apiVersionOpt.isPresent()) {
                 request.addHeader(LobClient.LOB_VERSION_HEADER, apiVersionOpt.get());
             }
 
+            try {
+                Properties prop = new Properties();
+                prop.load(AsyncLobClient.class.getClassLoader().getResourceAsStream("project.properties"));
+                wrapperVersion = prop.getProperty("version");
+            } catch (IOException e) {
+                wrapperVersion = "";
+            }
+
+            request.addHeader("user-agent", new StringBuilder().append("LobJava/").append(wrapperVersion).append(" JDK/").append(JAVA_VERSION).toString());
```

Comment:
Java will compile the overloaded `+` operator into a StringBuilder, so that would be instantiating 4 StringBuilders per request. I couldn't figured out how to load the value before the request is initiated since this is a static method.